### PR TITLE
TST: parametrize/de-duplicate arithmetic tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Review by mroeschke — APPROVED** [review]

File: pandas/core/indexes/range.py [hunk 1]
```diff
@@ -1041,7 +1041,11 @@ def _arith_method(self, other, op):
                 rstop = op(left.stop, right)
 
             res_name = ops.get_op_result_name(self, other)
-            result = type(self)(rstart, rstop, rstep, name=res_name)
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore", DeprecationWarning)
+                # The constructor validation can lead to a DeprecationWarning
+                #  from numpy, e.g. with RangeIndex + np.datetime64("now")
```
Comment (jreback) [inline pandas/core/indexes/range.py:1047]: umm shouldn't we do something about this? (e.g. cast to object)

File: pandas/tests/arithmetic/test_datetime64.py [hunk 2]
```diff
@@ -1220,13 +1233,21 @@ def test_dti_add_tick_tzaware(self, tz_aware_fixture, box_with_array):
...
         expected = tm.box_expected(expected, box_with_array)
 
-        # TODO: sub?
         for scalar in [pd.offsets.Hour(5), np.timedelta64(5, "h"), timedelta(hours=5)]:
             offset = dates + scalar
             tm.assert_equal(offset, expected)
             offset = scalar + dates
             tm.assert_equal(offset, expected)
 
+            roundtrip = offset - scalar
```
Comment (jreback) [inline pandas/tests/arithmetic/test_datetime64.py:1242]: prob should split this to a separate test & parameterize (followup)

**Review by jreback — APPROVED** [review]
lgtm some commments for followup

File: pandas/core/indexes/range.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/range.py:1047]: yes, turns out to be pretty easy, will handle before long

**Pull request merged**